[MINOR] : Code Cleanup - Misc modules

sjhajharia opened the pull request:
## What
Code Cleanup in Misc Modules

## Changes
Some common changes include

- Replace the Arrays.asList() with Collections.singletonList() wherever possible
- Cleaning up some if-else blocks
- Replacing some instances of String.builder() with String operations

Thanks!

chia7712 commented:
@sjhajharia Could you please fix the conflicts

sjhajharia commented:
Thanks @chia7712 !
Just merged the trunk and resolved the conflicts. Hopefully it gets reviewed this time before ti turns stale again! 😅

chia7712 commented on core/src/test/java/kafka/server/handlers/DescribeTopicPartitionsRequestHandlerTest.java:
```diff
@@ -105,12 +106,12 @@ public KafkaPrincipal deserialize(byte[] bytes) throws SerializationException {
-                .setSecurityProtocol(SecurityProtocol.PLAINTEXT.id)
-                .setListener(plaintextListener.value())
+        .setEndpoints(Collections.singletonList(
+                new UpdateMetadataRequestData.UpdateMetadataEndpoint()
```
Could you please revert those unrelated changes to reduce the size of PR?

chia7712 reviewed (commented):
@sjhajharia thanks for this cleanup. overall LGTM

edoardocomar committed:
KAFKA-16047: Use REQUEST_TIMEOUT_MS_CONFIG in AdminClient.fenceProducers (#16151)

Use REQUEST_TIMEOUT_MS_CONFIG in AdminClient.fenceProducers,
or options.timeoutMs if specified, as transaction timeout.

No transaction will be started with this timeout, but
ReplicaManager.appendRecords uses this value as its timeout.
Use REQUEST_TIMEOUT_MS_CONFIG like a regular producer append
to allow for replication to take place.

Co-Authored-By: Adrian Preston <prestona@uk.ibm.com>
